src/plugins/http.js

```diff
@@ -80,6 +81,28 @@ function patch (http, tracer, config) {
       pathname: options.path || options.pathname || '/'
     })
   }
+
+  let normalizeArgs
+  if (semver.satisfies(process.version, '>=10')) {
+    normalizeArgs = function normalizeArgs (inputURL, inputOptions, callback) {
+      let options = typeof inputURL === 'string' ? url.parse(inputURL) : Object.assign({}, inputURL)
+      options.headers = options.headers || {}
+      if (typeof inputOptions === 'object') {
+        options = Object.assign(options, inputOptions)
+      } else if (typeof inputOptions === 'function') {
+        callback = inputOptions
+      }
+      const uri = extractUrl(options)
+      return { uri, options, callback }
+    }
+  } else {
+    normalizeArgs = function normalizeArgs (options, callback) {
+      const uri = extractUrl(options)
+      options = typeof options === 'string' ? url.parse(uri) : Object.assign({}, options)
+      options.headers = options.headers || {}
+      return { uri, options, callback }
+    }
+  }
```

Comment:
Did it this way to cut down on runtime work.

Reply:
I don't think the performance gain of optimizing such an already fast operation warrants increased code complexity. Did you benchmark this?

Reply:
Nope. I’ll update 👍